src/common/collection.c

```diff
@@ -2031,7 +2031,7 @@ static gchar *get_query_string(const dt_collection_properties_t property, const
           g_free(name); // free the original filename
         }
 
-        if(g_list_length(list) > 0)
```

Comment:
We already switched most of the uses of `length() == 0` and `length() > 0` to bare NULL (non-NULL) checks a couple of years ago. It would be quite difficult to find all of them to put in an is_empty call instead.

Can do on moving the length_equal functions to utility.h.  Didn't at the time because I didn't find any other places where the lengths of two lists were compared.